File: credit.py
num = int(input("Number: "))
tmp = num
sum = 0
while tmp > 0:
    mod = tmp % 100
    t = mod
    tmp = int(tmp / 100)
    mod = int(mod / 10)
    mod = mod * 2
    while mod > 0:
        sum = sum + mod % 10
        mod = int(mod / 10)
    sum = sum + t % 10
#print(sum)
digit = 0;
tmp2 = num;
while tmp2 > 0:
    digit = digit + 1
    tmp2 = int(tmp2 / 10)

if sum % 10 != 0:
    print("INVALID\n")
else:
    #print("Hi\n")
    if digit == 15:
        if int(num / 10000000000000) == 34 or int(num / 10000000000000) == 37:
            print("AMEX\n")
        else:
            print("INVALID\n")
    elif digit == 13:
        if int(num / 1000000000000) == 4:
            print("VISA\n")
        else:
            print("INVALID\n")
    elif digit == 16:
        if int(num / 100000000000000) >= 51 and int(num / 100000000000000) <= 55:
            print("MASTERCARD\n")
        elif int(num / 1000000000000000) == 4:
            print("VISA\n")
        else:
            print("INVALID\n")
    else:
        print("INVALID\n")

File: week1_C/credit.c
#include <stdio.h>

int main(void)
{
    //input
    long long num;
    printf("Number: ");
    scanf("%lld", &num);
    //printf("%lld", num);
    long long tmp = num;
    long long sum = 0;
    while (tmp > 0)
    {
        long long mod = tmp % 100;
        long long t = mod;
        tmp /= 100;
        mod = mod / 10;
        mod *= 2;
        while (mod > 0)
        {
            sum += mod % 10;
            mod /= 10;
        }
        sum += t % 10;
    }

    int digit = 0;
    long long tmp2 = num;
    while (tmp2 > 0)
    {
        digit++;
        tmp2 /= 10;
    }
    //printf("%d", digit);

    if (sum % 10 != 0)
        printf("INVALID\n");
    else
    {
        if (digit == 15)
        {
            if (num / 10000000000000 == 34 || num / 10000000000000 == 37)
                printf("AMEX\n");
            else
                printf("INVALID\n");
        }
        else if (digit == 13)
        {
            if (num / 1000000000000 == 4)
                printf("VISA\n");
            else
                printf("INVALID\n");
        }
        else if (digit == 16)
        {
            if (num / 100000000000000 >= 51 && num / 100000000000000 <= 55)
                printf("MASTERCARD\n");
            else if (num / 1000000000000000 == 4)
                printf("VISA\n");
            else
                printf("INVALID\n");
        }
        else
            printf("INVALID\n");
    }
}
